source/simulator/parameters.cc

```diff
@@ -1167,6 +1167,12 @@ namespace aspect
       prm.declare_entry ("Names of fields", "",
                          Patterns::List(Patterns::Anything()),
                          "A user-defined name for each of the compositional fields requested.");
+      prm.declare_entry ("Types of fields", "",
```

Comment:
I think this should either be empty (meaning all are generic) or have exactly n_compositions entries, so I think the default here is ok.

Reply:
I prefer @gassmoeller's suggestion, because there are several tests where all fields are chemical, and it would be easier to set that in one word. I agree that generic should be the default.

Reply:
Also extending a parameter that has only a single entry into a list that is as long as number of compositional fields is something we do in a lot of material models. I do not recall another parameter that is empty, but interpreted as something else.

Reply:
Sure, sounds good.

Reply:
See my new comment. It's easier to write good error messages if we can preserve whether the user has actively chosen to make all fields *something*, or just used a default option.

Reply:
I am ok with that, but I dont see how you handle the case when the user does not specify anything at all. At the moment it will just not set any types, which I think is not what we want. I thought we always want to have this vector filled with n_compositional_fields entries, and if none are given they should all be `generic`.